pulsar-client-cpp/lib/Url.cc

```diff
@@ -88,6 +88,12 @@ const std::string& Url::file() const { return file_; }
 
 const std::string& Url::parameter() const { return parameter_; }
 
+std::string Url::hostPort() const {
```

Comment:
How returning const reference:-
const std::string& hostPort() const;

Reply:
To return a reference, we would have to store the `hostPort` as a member variable. Since we don't always need the `hostPort()` in `Url` class, I prefer to just create it on the fly when needed.